src/librustdoc/html/format.rs

```diff
@@ -172,7 +172,7 @@ impl clean::GenericParamDef {
         display_fn(move |f| match self.kind {
             clean::GenericParamDefKind::Lifetime => write!(f, "{}", self.name),
             clean::GenericParamDefKind::Type { ref bounds, ref default, .. } => {
-                f.write_str(&self.name)?;
+                f.write_str(&self.name.to_string())?;
```

Comment:
You might need a `&*` to get the Deref coercion.
```suggestion
                f.write_str(self.name.as_str())?;
```